src/braket/circuits/free_parameter_expression.py

```diff
@@ -0,0 +1,122 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+from typing import Dict
+
+from sympy import Expr, sympify
+
+
+class FreeParameterExpression:
+    """
+    Class 'FreeParameterExpression'
+    """
+
+    def __init__(self, expression: Expr):
+        """
+                Initializes a FreeParameterExpression. Best practice is to initialize using
+                FreeParameters and Numbers. Not meant to be initialized directly.
+
+                Args:
+                    expression (Expr): The expression to use.
+
+                Examples:
+                    >>> expression_1 = FreeParameter("theta") * FreeParameter("alpha")
+                    >>> expression_2 = 1 + FreeParameter("beta") + 2 * FreeParameter("alpha")
+        ="""
+        self._expression = expression
+
+    @property
+    def expression(self) -> Expr:
```

Comment:
...and here. It seems like we have a hard coupling with sympy.

Reply:
Maybe it should be private?

Reply:
I think instead of making it private, I will return `repr(self._expression)` to make it less dependent on the package.